src/main/java/com/zendesk/maxwell/RowMap.java

```diff
@@ -153,10 +161,22 @@ public String toJSON() throws IOException {
 		if ( this.txCommit )
 			g.writeBooleanField("commit", true);
 
-		writeMapToJSON("data", this.data, false);
-
-		if ( !this.oldData.isEmpty()) {
-			writeMapToJSON("old", this.oldData, true);
+		if ( this.excludeColumns == null) {
+			writeMapToJSON("data", this.data, false);
+			if ( !this.oldData.isEmpty()) {
+				writeMapToJSON("old", this.oldData, true);
+			}
+		}
+		else {
+			for (String column : this.excludeColumns) {
+				this.data.remove(column);
+				this.oldData.remove(column);
+			}
+			
+			if ( !this.data.isEmpty() )
+				writeMapToJSON("data", this.data, true);
+			if ( !this.oldData.isEmpty() )
+				writeMapToJSON("old", this.oldData, true);
```

Comment:
this code can be simplified; first iterate and remove the excluded columns, then execute the same code path -- there's no need to do different things if you have excluded columns or not.